kion/resource_project_enforcement.go

```diff
@@ -0,0 +1,326 @@
+package kion
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	hc "github.com/kionsoftware/terraform-provider-kion/kion/internal/kionclient"
+)
+
+func resourceProjectEnforcement() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages enforcement rules for projects to control service usage based on various criteria like" +
+			"spend limits and timeframe restrictions. .\n\n" +
+			"This resource allows for creating, reading, updating, and deleting project-specific enforcement settings.",
+		CreateContext: resourceProjectEnforcementCreate,
+		ReadContext:   resourceProjectEnforcementRead,
+		UpdateContext: resourceProjectEnforcementUpdate,
+		DeleteContext: resourceProjectEnforcementDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional, user-provided description of the enforcement.",
+			},
+			"timeframe": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"lifetime", "month", "year", "funding_source"}, false),
+				Description:  "Timeframe of the enforcement. Valid values are 'lifetime', 'month', 'year', 'funding_source'.",
+			},
+			"spend_option": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
I believe this is optional in the API, as it defaults to `spend`.

Reply:
Is this one of the times that even though technically it's optional, it still makes sense for them to have to pass something in?  It seems like a weird UX to assume the defaults are known.  But at the same time, it might be weird UX if they know the default.